### ISSUE TITLE
Could not create Tensorflow Graph: Invalid argument: No OpKernel was registered to support Op 'Const' with these attrs

### ISSUE THREAD

**Issue by ashokbugude** [Qualeams/Android-Face-Recognition-with-Deep-Learning-Test-Framework#21]
I have built Tensoflow using bazel.
I have taken the libraries from tensorflow and used in android ndk, by stetting abi as either of 'arm64-v8a' or 'armeabiv-7a'. Not Both specified together
tensorflow::Status s = session->Create(tensorflow_graph); 

Below are the debug messages

06-03 15:51:03.191 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:113 Loading Tensorflow.
06-03 15:51:03.191 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:115 Making new SessionOptions.
06-03 15:51:03.191 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:118 Got config, 0 devices
06-03 15:51:03.193 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:121 Session created.
06-03 15:51:03.193 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:124 Graph created.
06-03 15:51:03.193 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:128 Acquired AssetManager.
06-03 15:51:03.193 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:130 Reading file to proto: /storage/emulated/0/Pictures/facerecognition/data/TensorFlow/vgg_faces.pb
06-03 15:51:05.045 25365-26625/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:135 Creating session.
06-03 15:51:05.068 25365-26625/ch.zhaw.facerecognition E/native: tensorflow_jni.cc:138 Could not create Tensorflow Graph: Invalid argument: No OpKernel was registered to support Op 'Const' with these attrs. Registered devices: [CPU], Registered kernels:

**Comment by sladomic** [comment]
@ashokbugude which model do you use? By default TensorFlow builds the libs for ARM devices with less operations support, because they usually only need operations for inference and not for training. So in the freezing of the model you need to get rid of these operations or you need to change the settings of TensorFlow when building for ARM devices. See also https://petewarden.com/2016/09/27/tensorflow-for-mobile-poets/ or https://github.com/davidsandberg/facenet/pull/240 which I used to get a suitable facenet model (see my last commits)

**Comment by ashokbugude** [comment]
I have used the model vggFaces.pbb as described in your earlier version of readme.md file

**Comment by sladomic** [comment]
can you try the optimized_facenet.pb model with the following settings:
N = 160
Input Size = 160
Output SIze = 128
Input layer = input
Output layer = embeddings
Model file = optimized_facenet.pb

**Comment by ashokbugude** [comment]
Can I know where to get the model file from ?
Is it also 100% accurate as mentioned by you for vggFaces.pb ?

**Comment by sladomic** [comment]
https://github.com/Qualeams/Android-Face-Recognition-with-Deep-Learning-Library/blob/master/facerecognitionlibrary/src/main/assets/optimized_facenet.pb
the accuracy with our test set (6 persons with 100 images per person for training and 100 for test) was also 100%, yes.

**Comment by ashokbugude** [comment]
Hi 
I am receiving the same error for 'armeabi-v7a'  or 'arm64-v8a'  when using optimized_facenet.pb as shown below

06-03 18:42:00.061 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:113 Loading Tensorflow.
06-03 18:42:00.061 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:115 Making new SessionOptions.
06-03 18:42:00.061 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:118 Got config, 0 devices
06-03 18:42:00.069 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:121 Session created.
06-03 18:42:00.069 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:124 Graph created.
06-03 18:42:00.069 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:128 Acquired AssetManager.
06-03 18:42:00.069 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:130 Reading file to proto: /storage/emulated/0/Pictures/facerecognition/data/TensorFlow/optimized_facenet.pb
06-03 18:42:00.327 22671-23306/ch.zhaw.facerecognition I/native: tensorflow_jni.cc:135 Creating session.
06-03 18:42:00.363 22671-23306/ch.zhaw.facerecognition E/native: tensorflow_jni.cc:138 Could not create Tensorflow Graph: Invalid argument: No OpKernel was registered to support Op 'Const' with these attrs.  Registered devices: [CPU], Registered kernels:
                                                                   <no registered kernels>
                                                                 
                                                                 	 [[Node: InceptionResnetV1/Block8/Branch_0/Conv2d_1x1/BatchNorm/batchnorm/sub/_376__cf__376 = Const[dtype=DT_FLOAT, value=Tensor<type: float shape: [192] values: -0.082128763 -0.060227871 0.04927659...>]()]]

**Comment by sladomic** [comment]
Hmm, strange. Which version of TF do you use?

Try to change the default settings for the included operations when building for ARM devices. See https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile

**Comment by ashokbugude** [comment]
I am ussing the latest version of Tensor Flow.
I havn't made below seetings 
N = 160
Input Size = 160
Output SIze = 128
Input layer = input
Output layer = embeddings
Model file = optimized_facenet.pb
Will make these and try again and shall follow the above link

**Comment by ashokbugude** [comment]
I have followed the steps in https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile
But the error remains

The output of the above link is a graph loaded into my mobile
Do I need to make any changes in code of tensorflow_jni.cc , because the current code does not detect/read from the graph loaded in mobile

**Comment by ashokbugude** [comment]
Its working fine with older version of Tensor Flow and vgg_faces.pb as described in https://github.com/Qualeams/Android-Face-Recognition-with-Deep-Learning-Library/issues/1
